DependencyInjection/Compiler/AddProcessorsPass.php

```diff
@@ -46,13 +48,15 @@ public function process(ContainerBuilder $container)
                     $definition = $container->getDefinition('monolog.logger_prototype');
                 }
 
-                if (!empty($tag['method'])) {
-                    $processor = [new Reference($id), $tag['method']];
-                } else {
-                    // If no method is defined, fallback to use __invoke
-                    $processor = new Reference($id);
+                if (1 === Logger::API || is_a($definition->getClass(), ProcessableHandlerInterface::class, true)) {
+                    if (!empty($tag['method'])) {
+                        $processor = [new Reference($id), $tag['method']];
+                    } else {
+                        // If no method is defined, fallback to use __invoke
+                        $processor = new Reference($id);
+                    }
+                    $definition->addMethodCall('pushProcessor', [$processor]);
                 }
```

Comment:
With Monolog v2, should we throw Exception if handler doesn't implements `ProcessableHandlerInterface` here?